### ISSUE TITLE
Gstatus '-' stops working over time

### ISSUE THREAD

**Issue by Nomad145** [tpope/vim-fugitive#779]
Greetings,

I have an issue with adding files in `Gstatus` after working on a repo for a while .  When I first cloned the repo, I was able to add files with the `-` key but after a month or so it has stopped working.  It seems like everything else still works, as I can `Gcommit`, `Glog`, and I can even do `D` within `Gstatus`.

I cloned the repo from the same folder the project was in (`git clone folder new_folder`) and `Gstatus` works for that directory.  Its also worth noting that the Syntax Highlighting isn't set within `Gstatus` for the repo that has the issue and is set for the repo that works fine.

Any ideas what the issue might be?

**Comment by tpope** [comment]
Check `:verbose set filetype?`

**Comment by Nomad145** [comment]
```
  filetype=php
        Last set from /usr/share/vim/vim73/filetype.vim
```

**Comment by tpope** [comment]
Well there's your problem.

**Comment by Nomad145** [comment]
I realize what you meant now.  I did `:verbose set filetype` inside `:Gstatus` and it shows:
```
  filetype=gitcommit
        Last set from ~/Code/maximum-awesome/vim/bundle/vim-fugitive/plugin/fugitive.vim
```

**Comment by justinmk** [comment]
@Nomad145 Maybe the filetype detection in Vim 7.3 is buggy. For PHP, it looks for strings in the file to detect the file type. There must be a string in your `:Gstatus` buffer that leads `filetype.vim` to conclude it is a PHP file.

**Comment by Nomad145** [comment]
My first comment was the filetype detected by Vim within a buffer that was a PHP file.  The second was the filetype within the `:Gstatus` buffer.  I wasn't sure exactly what @tpope was asking.

It seems to be detecting the filetype properly as I have compared the detected filetypes within a copy of the repo that works as well as a copy that does not and they are the same.

Here are the outputs from both the copy of the repo that works as well as the one that doesn't.

**Non-Working Copy**
```
filetype=gitcommit
        Last set from ~/Code/maximum-awesome/vim/bundle/vim-fugitive/plugin/fugitive.vim
```

**Working Copy**
```
  filetype=gitcommit
        Last set from ~/Code/maximum-awesome/vim/bundle/vim-fugitive/plugin/fugitive.vim
```

The output for both is within `:Gstatus`.  I opened a markdown file this time.

If its not a Fugitive issue, any direction on what it might be?

**Comment by tpope** [comment]
If the file type is right, but the syntax highlighting is wrong, that almost always means you've got another plugin in the mix doing something weird. Buffer managers and window managers are popular culprits.

**Comment by Nomad145** [comment]
I'll see if I can figure out what exactly it is, then.  Thanks for the help!

**Comment by Nomad145** [comment]
Finally got around to debugging this issue.  I have a blank `vim.rc` using only `vim-fugitive` and I'm still having issues with not being able to stage commits.

Here's the output of my `:scriptnames`:

```
  1: ~/blankrc
  2: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/syntax.vim
  3: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/synload.vim
  4: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/syncolor.vim
  5: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/filetype.vim
  6: ~/Code/maximum-awesome/vim/plugin/vim-fugitive/plugin/fugitive.vim
  7: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/getscriptPlugin.vim
  8: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/gzip.vim
  9: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/logiPat.vim
 10: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/manpager.vim
 11: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/matchparen.vim
 12: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/netrwPlugin.vim
 13: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/rrhelper.vim
 14: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/spellfile.vim
 15: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/tarPlugin.vim
 16: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/tohtml.vim
 17: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/vimballPlugin.vim
 18: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/plugin/zipPlugin.vim
 19: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/php.vim
 20: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/html.vim
 21: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/javascript.vim
 22: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/vb.vim
 23: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/css.vim
 24: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/sql.vim
 25: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/sqloracle.vim
 26: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/scripts.vim
 27: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/gitcommit.vim
 28: /usr/local/Cellar/vim/8.0.0002/share/vim/vim80/syntax/diff.vim
```

Here's my blank `vim.rc`:

```
set nocompatible
syntax enable
```

Any ideas where to look next?  I've upgraded to Vim 8, but I can test for 7.4 if needed.

**Comment by h3xx** [comment]
Suggest closing this issue:

* User unable or unwilling to reproduce issue in a clean-room environment.
* Discussion has strayed into tech-support for this user's specific setup, rather than addressing the bug itself.

**Comment by tpope** [comment]
Looks clean room to me? Though "still having issues" needs elaboration.

**Comment by Nomad145** [comment]
I found the problem.  Fugitive matches the output of `git status -c status.displayCommentPrefix=true` with [^# .*:$](https://github.com/tpope/vim-fugitive/blob/master/plugin/fugitive.vim#L3080).

If you use issue tracker slugs with a `#` prefix and write your commit message interactively, Git's `core.commentchar` must configured with another character.  In my case, I used `core.commentchar = *`.  This breaks the regex in `foldtext()`.  Unsetting `core.commentchar` solves the issue.
